python-demo/strategy/funding_arbitrage/funding_arbitrage.py

```diff
@@ -0,0 +1,351 @@
+    def __init__(self, setting, logger, mainnet_configs, testnet_configs):
+        super().__init__(setting, logger, mainnet_configs, testnet_configs)
+        self.setting = setting
+        self.is_trading = False
+        self.init_strategy()
+
+    def init_strategy(self):
+        self.ap1 = self.bp1 = self.binance_ap1 = self.binance_bp1 = -1
+        self.av1 = self.bv1 = -1
+        self.curr_duration_volume = 0
+        self.last_duration_volume = 0
+        self.tick = None
+
+        self.inj_funding_rate = 0.0
+        self.binance_funding_rate = 0.0
+        self.net_position = 0
+        self.net_position_binance = 0
+
+        self.arb_threshold = float(self.setting["arb_threshold"])
+        self.interval = int(self.setting["interval"])
+        self.binance_api_key = self.setting["binance_api_key"]
+        self.binance_api_secret = self.setting["binance_api_secret"]
+        self.active_orders = {}  # [order_hash, : order_data]
+
+        self.order_size = float(self.setting["order_size"])
+        self.gas_price = 500000000
+        self.strategy_name = self.setting["strategy_name"]
+        if self.setting.get("start_time", None):
+            self.start_time = datetime.strptime(self.setting["start_time"])
+        else:
+            self.start_time = datetime.utcnow()
+            self.setting["start_time"] = self.start_time.strftime(
+                "%Y-%m-%d %H:%M%S.%f")
+
+        self.add_schedule()
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(self.get_init_position())
+        loop.run_until_complete(
+            self.get_open_orders(self.acc_id, self.market_id))
+        loop.run_until_complete(self.get_orderbook())
+        self.msg_list = []
+
+        self.subscribe_stream()
+        self.logger.debug("finish init")
+
+    def add_schedule(self):
+        self.sched = AsyncIOScheduler()
+        self.sched.add_job(self.on_timer, 'interval',
+                           seconds=self.interval)
+
+    def subscribe_stream(self):
+        self.tasks = [
+            asyncio.Task(self.stream_order(self.market_id, self.acc_id)),
+            asyncio.Task(self.stream_trade(self.market_id, self.acc_id)),
+            asyncio.Task(self.stream_position(self.market_id, self.acc_id)),
+            asyncio.Task(self.stream_orderbook(self.market_id)),
+            asyncio.Task(self.subscribe_binance_orderbook()),
+            asyncio.Task(self.subscribe_inj_exchange_funding()),
+            asyncio.Task(self.subscribe_binance_exchange_funding()),
+            asyncio.Task(self.subscribe_binance_position())
+        ]
+
+    async def subscribe_inj_exchange_funding(self):
+        markets = await self.client.stream_derivative_markets()
+        async for market in markets:
+            # print(market)
+            if market.market.market_id == self.market_id:
+                self.inj_funding_rate = float(
+                    market.market.perpetual_market_info.hourly_funding_rate_cap)
```

Comment:
This is the funding rate cap, the max funding a market can have at a given time. The Exchange API doesn't return the current funding rate at the moment this will be added in the future but for now we need to calculate it manually.